superset-frontend/src/components/Select/utils.ts

```diff
@@ -61,7 +61,7 @@ export function findValue<OptionType extends OptionTypeBase>(
 }
 
 export function getValue(option: string | number | { value: string | number }) {
-  return typeof option === 'object' ? option.value : option;
+  return option && typeof option === 'object' ? option.value : option;
```

Comment:
If we are allowing `option` to be `null` then we need to update the signature of `getValue` as well (add `| null`).